extensions/wikia/TemplateClassification/TemplateClassification.hooks.php

```diff
@@ -15,7 +14,96 @@ public static function register() {
 		\Hooks::register( 'BeforePageDisplay', [ $hooks, 'onBeforePageDisplay' ] );
 		\Hooks::register( 'PageHeaderPageTypePrepared', [ $hooks, 'onPageHeaderPageTypePrepared' ] );
 		\Hooks::register( 'QueryPageUseResultsBeforeRecache', [ $hooks, 'onQueryPageUseResultsBeforeRecache' ] );
+		/* Edit page hooks */
+		\Hooks::register( 'EditPage::showEditForm:fields', [ $hooks, 'onEditPageShowEditFormFields' ] );
+		\Hooks::register( 'ArticleInsertComplete', [ $hooks, 'onArticleInsertComplete' ] );
 		\Hooks::register( 'EditPageLayoutExecute', [ $hooks, 'onEditPageLayoutExecute' ] );
+		\Hooks::register( 'EditPageMakeGlobalVariablesScript', [ $hooks, 'onEditPageMakeGlobalVariablesScript' ] );
+	}
+
+	/**
+	 * Save template type passed from article creation
+	 * template type is stored in templateClassificationType hidden field
+	 *
+	 * @param \WikiPage $wikiPage
+	 * @return bool
+	 */
+	public function onArticleInsertComplete( \WikiPage $wikiPage ) {
+		global $wgCityId;
+
+		$request = \RequestContext::getMain()->getRequest();
+		$type = $request->getVal( 'templateClassificationType' );
+
+		/**
+		 * The service was not available when the field's value was set
+		 * so we exit early to prevent polluting of the results.
+		 */
+		if ( $type === \TemplateClassificationService::NOT_AVAILABLE ) {
+			return true;
+		}
+
+		try {
+			( new \TemplateClassificationService() )->classifyTemplate(
+				$wgCityId,
+				$wikiPage->getId(),
+				$type,
+				\TemplateClassificationService::USER_PROVIDER,
+				$wikiPage->getUser()
+			);
+		} catch ( \Exception $e ) {
```

Comment:
Cool. :) For reference though, if you need to, you can catch multiple Exception types, i.e. with either a catch block for each exception type, or if you really want to avoid code duplication, to hack around PHP lacking the ability to catch multiple exception types in one block, you can catch the generic exception and then use `instanceof` to check if the exception is a type you want to catch and rethrow it if it isn't.